mh/pat.py

```diff
@@ -1230,39 +1230,72 @@ def sendAnsUserSearchInfo(self, capcom_id, search_info, seq):
         TR: User search data response
         """
         users = self.session.find_users(capcom_id, "", 1, 1)
-        assert users, "Capcom ID not found"
-        user = users[0]
-
-        user_info = pati.UserSearchInfo()
-        user_info.capcom_id = pati.String(user.capcom_id)
-        user_info.hunter_name = pati.String(user.hunter_name)
-        user_info.stats = pati.Binary(user.hunter_info.pack())
-        user_info.layer_host = pati.Binary(user.get_layer_host_data())
-
-        user_info.unk_byte_0x07 = pati.Byte(1)
-        user_info.server_name = pati.String("\t".join([
-            user.local_info["server_name"] or "",
-            user.local_info["gate_name"] or "",
-            user.local_info["city_name"] or ""
-        ]))
-        user_info.unk_byte_0x0b = pati.Byte(1)
-        user_info.unk_string_0x0c = pati.String("StrC")
-        user_info.city_capacity = pati.Long(4)
-        user_info.city_size = pati.Long(3)
-
-        # This fields are used to identify a user.
-        # Specifically when a client is deserializing data from the packets
-        # `NtcLayerBinary` and `NtcLayerBinary2`
-        # TODO: Proper field value and name
-        user_info.info_mine_0x0f = pati.Long(int(hash(user.capcom_id))
-                                             & 0xffffffff)
-        user_info.info_mine_0x10 = pati.Long(int(hash(user.capcom_id[::-1]))
-                                             & 0xffffffff)
-
-        data = user_info.pack()
-        # TODO: Figure out the optional fields
-        data += pati.pack_optional_fields([])
-        self.send_packet(PatID4.AnsUserSearchInfo, data, seq)
+        if not users:
+            # User is offline or doesn't exist
+            basic_user_info = self.session.find_capcom_id(capcom_id)
+            hunter_info = self.session.fetch_hunter_info(capcom_id)
+            assert basic_user_info is not None, "Capcom ID not found"
+            assert hunter_info is not None, "Hunter information not found"
+
+            search_info.capcom_id = pati.String(capcom_id)
+            search_info.hunter_name = pati.String(basic_user_info['name'])
+            data = search_info.pack()
+
+            # TODO: Figure out the optional fields
+            # field0_0x0, unk
+            # field1_0x1, unk
+            # field4_0x4, unk
+            data += pati.pack_optional_fields([
```

Comment:
I think [this](https://github.com/sepalani/MH3SP/blob/dev/mh/session.py#L315) are the (incomplete) optional field that you are looking fuzzing.

Reply:
This one would require actual database storage of player objects, so out of scope for this PR